core/dns.go

```diff
@@ -79,47 +103,37 @@ func (dnsResolver *DNSResolverImpl) LookupTXT(hostname string) ([]string, time.D
 	return txt, rtt, err
 }
 
-// LookupHost sends a DNS query to find all A/AAAA records associated with
-// the provided hostname.
-func (dnsResolver *DNSResolverImpl) LookupHost(hostname string) ([]net.IP, time.Duration, time.Duration, error) {
-	var addrs []net.IP
-	var answers []dns.RR
+func isPrivateV4(ip net.IP) bool {
+	return privateNetworkA.Contains(ip) || privateNetworkB.Contains(ip) || privateNetworkC.Contains(ip)
+}
 
-	r, aRtt, err := dnsResolver.ExchangeOne(hostname, dns.TypeA)
-	if err != nil {
-		return addrs, 0, 0, err
-	}
-	if r.Rcode != dns.RcodeSuccess {
-		err = fmt.Errorf("DNS failure: %d-%s for A query", r.Rcode, dns.RcodeToString[r.Rcode])
-		return nil, aRtt, 0, err
-	}
+func isPrivateV6(ip net.IP) bool {
+	return privateNetworkD.Contains(ip)
+}
 
-	answers = append(answers, r.Answer...)
+// LookupHost sends a DNS query to find all A records associated with the provided
+// hostname.
+func (dnsResolver *DNSResolverImpl) LookupHost(hostname string) ([]net.IP, time.Duration, error) {
```

Comment:
This method needs to follow at least CNAME redirection, ideally also DNAME.  That is a very common way of redirecting things for HTTP servers, so if we don't support it, it will cause a large degree of breakage.